session_integration_test.go

```diff
@@ -230,3 +233,112 @@ func TestSessionIterIntegration(t *testing.T) { // nolint:paralleltest // Integr
 		}
 	}
 }
+
+var (
+	caPath   = "testdata/tls/cadb.pem"
+	certPath = "testdata/tls/db.crt"
+	keyPath  = "testdata/tls/db.key"
+)
+
+func TestTLSIntegration(t *testing.T) {
+	testCases := []struct {
+		name      string
+		tlsConfig *tls.Config
+		caPath    string
+		certPath  string
+		keyPath   string
+	}{
+		{
+			name:      "no tls",
+			tlsConfig: nil,
+		},
+		{
+			name: "tls - no client verification",
+			tlsConfig: &tls.Config{
+				RootCAs:            x509.NewCertPool(),
+				InsecureSkipVerify: true,
+			},
+			certPath: certPath,
+			keyPath:  keyPath,
+		},
+		{
+			name: "tls - with client verification",
+			tlsConfig: &tls.Config{
+				RootCAs:            x509.NewCertPool(),
+				InsecureSkipVerify: false,
```

Comment:
Out of curiosity: how did you make it pass correctly with client verification - yesterday it wasn't working, what was the problem?

Reply:
Without setting RootCA to DB CA you shouldn't be able to verify certificate, something is fishy :)

Reply:
> Out of curiosity: how did you make it pass correctly with client verification - yesterday it wasn't working, what was the problem?

The problem was that if InsecureSkipVerify flag is false ServerName must be provided and supplying node's IP address as ServerName wouldn't work with certificates from gocql repo as they don't list any SAN's. I created new certificates with node's IP listed as a SAN, with it it works. I guess it has to do with the fact that default gocql sslOptions default to not doing client verification (setting InsecureSkipVerifyFlag to true).